src/convertToObject.js

```diff
@@ -3,16 +3,30 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
- * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
- *
+
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styleObject = {};
+  const sourceArr = sourceString
+    .split(';\n')
+    .filter((string) => string.includes(':'));
+
+  sourceArr.forEach((style) => {
+    const styleArray = style.split(':');
+    const key = styleArray[0].trim();
+    const styleValueSet = new Set(styleArray[1].split(' '));
+
+    styleValueSet.delete('');
+
+    const styleValueArr = [...styleValueSet];
```

Comment:
There is no need to convert the set back into an array and then join it into a string. The join method can be applied directly to the array resulting from the split method at line 23.